corehq/motech/repeaters/models.py

```diff
@@ -264,10 +264,10 @@ class Repeater(RepeaterSuperProxy):
         default=REQUEST_POST,
         max_length=16,
     )
-    is_paused = models.BooleanField(default=False)
-    next_attempt_at = models.DateTimeField(null=True, blank=True)
+    is_paused = models.BooleanField(default=False, db_index=True)
+    next_attempt_at = models.DateTimeField(null=True, blank=True, db_index=True)
     last_attempt_at = models.DateTimeField(null=True, blank=True)
-    # TODO: max_workers = models.IntegerField(default=1)
+    max_workers = models.IntegerField(default=MAX_REPEATER_WORKERS)
```

Comment:
What do you think of changing this default to zero and changing the `num_workers` calculation below to `self.max_workers or MAX_REPEATER_WORKERS`? That way if we change the constant in code it will not require a database schema change as well.

I assume it will be rare for the max workers to be overridden on a repeater. Is that a good assumption? In fact, I'm curious when we ever expect to override the code constant with a database value? Currently it looks like a field that is not set anywhere in code, which implies that it is a custom setting for developers. Do I have that right?

Reply:
> What do you think of changing this default to zero and changing the `num_workers` calculation below to `self.max_workers or MAX_REPEATER_WORKERS`?

Excellent idea.

> In fact, I'm curious when we ever expect to override the code constant with a database value?

I am thinking of two scenarios:
1. The ability to ensure that payloads are always send in chronological order
2. Remote endpoints that are unable to support `MAX_REPEATER_WORKERS` concurrent requests

At first I thought that a repeater should not be allowed to exceed the maximum. But if we rename `MAX_REPEATER_WORKERS` to `DEFAULT_REPEATER_WORKERS` that could better communicate that a repeater can exceed that value, if the endpoint can handle large numbers of workers. A high-volume repeater sending to a service built for that kind of volume (like Zapier) would be fine.

Reply:
eb1f4245969

That's a fixup commit. I'd like to rebase before flipping this to a normal PR, to clean up the reverted commits and the migrations (which will probably conflict with master soon).